src/operators/AxisAlignedSlice4D/PyAxisAlignedSlice4DAttributes.C

```diff
@@ -158,6 +158,11 @@ AxisAlignedSlice4DAttributes_SetI(PyObject *self, PyObject *args)
                 PyErr_Clear();
                 return PyErr_Format(PyExc_TypeError, "arg %d not interpretable as C++ int", (int) i);
             }
+            if (fabs(double(val))>1.5E-7 && fabs((double(long(cval))-double(val))/double(val))>1.5E-7)
```

Comment:
Why is the error checking logic here different than the if-section just preceding (starting at line 127)?

Reply:
The error check above is checking if `PyLong_AsLong()` failed entirely. This line checks if the value returned by the Python conversion call fails to reprsent, faithfully, the same value in the intended destination C/C++ variable. Some good examples `PyLong_AsLong()` (the type involved here anyways) converts `123123123` fine, but the destination C/C++ variable is unsigned char. That would fail the `fabs()` test logic. We use `fabs()` with machine precisioin (1.5E-7) because explicit equality tests don't work for float/double and the macro code block this all comes from is intended to work with *any* type.

Reply:
I am saying that there is more error-checking in one if-clause (the one staring on line 139) than the other (the one starting on line 127).  If all this extra error checking is needed, then it should be in both if-clauses.

Reply:
Sorry. Not following. I don't agree. One if clause is testing for one kind of failure. The other is testing for a different kind of failure. Maybe the fact that both wind up generating the identical error message is what's creating problems for you? Otherwise, an example rewrite from you of this specific block would help me understand what you are saying.

Reply:
Wait a minute...I think I may see what you are saying. Let me go look at the macro that generates this.

Reply:
Doh! 🤦🏻 I see it. Thanks for catching this. There were a handful or cases in the macros using some older equality logic (e.g. `cval != val`) that I didn't update with this newer `fabs()` based logic. This will likely cause a significant regen. Will let you know when done.

Reply:
@biagas does this look like it makes more sense now...

https://github.com/visit-dav/visit/blob/629c89e177e591d3166c28ff9f5bde4e66ff053d/src/operators/AxisAlignedSlice4D/PyAxisAlignedSlice4DAttributes.C#L121-L182

Reply:
Yes, thanks.